### PR TITLE
Fixes #7525 - don't fail on structured facts

### PR DESCRIPTION
**Pull request by lzap**

### CODE REVIEW

**Comment by dLobatog** [comment]
@lzap Looks good, just add a test for these 'broken' facts and we're all set :smile:

**Comment by lzap** [comment]
Added a test for numbers as it's the most important one today. We pretty much convert everything via `to_s` but I don't think arrays or nested hashes are worth the effort (we are changing this soon).

File: test/unit/fact_importer_test.rb [hunk 1]
```diff
@@ -10,6 +10,12 @@ class FactImporterTest < ActiveSupport::TestCase
     assert_equal PuppetFactImporter, FactImporter.importer_for('whatever')
   end
 
+  test "normalize converts to strings" do
+    facts = { "stringfact" => "test", "integerfact" => 42, "arrayfact" => ['test'] }
+    expected = { "stringfact" => "test", "integerfact" => "42", "arrayfact" => "['test']" }
+    assert_equal expected, FactImporter.normalize(facts)
```
Comment (dLobatog) [inline test/unit/fact_importer_test.rb]: @lzap `normalize` is not a class method, so maybe `FactImporter.new(nil, facts)` is what you want to do, if tests are :green_heart: after that, I'll merge!

Comment (lzap) [inline test/unit/fact_importer_test.rb]: > It's not a class method, so maybe `FactImporter.new(nil, facts)` is what you want to do, if tests are :green_heart: after that, I'll merge!

Oh I did not mean to add this one, I thought this bit was deleted. I
have removed this as the method is fully covered in the puppet importer
already.

## 

Later,
 Lukas #lzap Zapletal

**Comment by dLobatog** [comment]
Merged as 2ec9c96fb577171eeb850b8847d5fd5f044ec4e0, thanks @lzap !

**Pull request closed without merging**